public/javascript/scripts.js

```diff
@@ -0,0 +1,278 @@
+    var totalHtml = "";
+    
+    data.forEach(function(item, index) {
+        html = defaultAssetRow;
+        html = html.replace(/TAG/g, item.tag + index);
+        html = html.replace(/PARENT/g, item.parent_tag);
+        html = html.replace(/CUSTOMER/g, item.customer);
+        html = html.replace(/RECEIVED/g, item.received);
+        html = html.replace(/CUST_AG/g, item.customer_tag);
+        html = html.replace(/SERIAL/g, item.serial);
+        html = html.replace(/ASSET_TYPE/g, item.asset_type);
+        html = html.replace(/MANUFACTURER/g, item.manufacturer);
+        html = html.replace(/PRODUCT/g, item.product);
+        html = html.replace(/MODEL/g, item.model);
+        html = html.replace(/LOCATION/g, item.location);
+        
+        totalHtml += html;
+    });
+    
+    $("#assetTableBody").html(totalHtml);
+}
+
+function showAssetInfo(asset) {
+    $("#noAssetMessage").hide();
+    var endHtml = "";
+    for (var property in asset) {
+        if (asset.hasOwnProperty(property)) {
+            var html = ""; //the html for a single property yeuah
+            switch(property) {
+                case 'tag' :
+                    html += assetInfoNoChange;
+                    break;
+                case 'customer':
+                    html += assetInfoSelect;
+                    break;
+                case 'received':
+                    html += assetInfoDate;
+                    var date = new Date(asset[property]);
+                    html = html.replace(/VALUE/g, date.toISOString().substring(0, date.toISOString().indexOf('T')));
+                    break;
+                case 'asset_type' :
+                    html += assetInfoSelect;
+                    break;
+                case 'sold_date' :
+                    html += assetInfoDate;
+                    var date = new Date(asset[property]);
+                    html = html.replace(/VALUE/g, date.toISOString().substring(0, date.toISOString().indexOf('T')));
+                    break;
+                case 'price' :
+                    html += assetInfoNumber;
+                    break;
+                case 'sold_via' :
+                    html += assetInfoSelect;
+                    break;
+                default :
+                    html += assetInfoText;
+                    break;
+            }
+            html = html.replace(/CTITLE/g, (property.substring(0, 1).toUpperCase() + property.substring(1)).replace('_', " "));
+            html = html.replace(/TITLE/g, property);
+            html = html.replace(/VALUE/g, asset[property]);
+            
+            endHtml += html;
+        }
+    }
+    
+    $("#assetInfo").html(endHtml);
+    
+    //THESE LISTENERS are here because they only apply to elemtents that exist on the page already
+    $("select").on('change', function() {
+        $("#" + $(this).attr('id').substring($(this).attr('id').indexOf('_') + 1)).val($(this).val());
+    });
+    
+    $("input").keypress(function(e){
```

Comment:
I do not like having all of the action listeners defined at the bottom of this function, but it is necessary because they must be defined AFTER each input type is added to the html. Therefore, each time the input fields change the lsiteners must be re-defined.